### PR TITLE
Update math functions

### PR DESCRIPTION
**Pull request by neon60**
This PR focusing on math functions update and those changes are in kernel_language.rst.

I based my changes on PR: #3474 This will be merged in soon at Gerrit side.

### CODE REVIEW

**Review by randyh62 — DISMISSED** [review]
Seems like some of the images in common between understand/hardware_implementation and understand/programming_model could be consolidated to use a single reference? It seems like this is being done in the kernel_language.rst file.

**Comment by neon60** [comment]
> Seems like some of the images in common between understand/hardware_implementation and understand/programming_model could be consolidated to use a single reference? It seems like this is being done in the kernel_language.rst file.

The develop branch is slightly behind, this PR changes are focusing on kernel_language.rst.

Because of the data structure contribution rule. We have file duplication, or we do not state how to handle those.

File: docs/reference/kernel_language.rst [hunk 1]
```diff
@@ -327,441 +327,441 @@ Following is the list of supported single precision mathematical functions.
...
+    * - | ``bool signbit(float  a)``
         | Return the sign bit of the input.
       - ✓
       - ✓
 
-    * - | void sincosf ( float  x, float* sptr, float* cptr )
+    * - | ``void sincosf(float  x, float* sptr, float* cptr)``
         | Calculate the sine and cosine of the first input argument.
       - ✓
       - ✗
```
Comment (matyas-streamhpc) [inline docs/reference/kernel_language.rst]: Can you check whether it is supported now?

It seems from this page that it is supported now, but I did not check it.
https://cgmb-hip.readthedocs.io/en/latest/autotoc_md76.html

Thanks!

File: docs/reference/kernel_language.rst [hunk 2]
```diff
@@ -1372,255 +1462,255 @@ HIP supports the following atomic operations.
...
+    * - ``double atomicAdd_system(double* address, double val)``
       - ✓
       - ✓
 
-    * - float unsafeAtomicAdd(float* address, float val)
+    * - ``float unsafeAtomicAdd(float* address, float val)``
       - ✓
       - ✗
 
-    * - float safeAtomicAdd(float* address, float val)
```
Comment (matyas-streamhpc) [inline docs/reference/kernel_language.rst:1419]: missing apostrophes:
` * - ``float safeAtomicAdd(float* address, float val)`` `

File: docs/reference/kernel_language.rst (hunk 1 above)
Comment (neon60) [reply docs/reference/kernel_language.rst:567]: Supported.

Comment (neon60) [reply docs/reference/kernel_language.rst]: I have to check this, thanks for the heads up.

File: docs/reference/kernel_language.rst (hunk 2 above)
Comment (neon60) [reply docs/reference/kernel_language.rst:1419]: Fixed. Thanks

File: docs/reference/kernel_language.rst (hunk 1 above)
Comment (neon60) [reply docs/reference/kernel_language.rst]: Fixed.

**Pull request merged**